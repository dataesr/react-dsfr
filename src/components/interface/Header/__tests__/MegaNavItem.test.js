import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { v4 as uuidv4 } from 'uuid';
import { MegaNavItem } from '../index';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('uuid', () => ({
  v4: jest.fn(),
}));

describe('<MegaNavItem />', () => {
  let wrapper;

  const initialProps = {
    title: 'title',
    link: 'link',
    children: 'child',
    linkLabel: 'label',
    current: false,
    as: 'h6',
    description: 'description',
    closeButtonLabel: 'close',
  };

  beforeEach(() => {
    uuidv4.mockImplementationOnce(() => 'xyx');
    wrapper = (props = {}, children) => shallow(
      <MegaNavItem
        {...props}
      >
        {children || props.children}
      </MegaNavItem>,
    );
  });
  it('renders correctly', () => {
    const component = renderer
      .create(
        <MegaNavItem {...initialProps} />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should have button with title as text', () => {
    const component = wrapper(initialProps);
    expect(component.find('.fr-nav__item > button').text()).toBe(initialProps.title);
  });

  it('should expand', () => {
    const component = wrapper(initialProps);
    component.find('.fr-nav__item > button').simulate('click');
    expect(component.find('.fr-collapse--expanded').exists()).toBeTruthy();
  });
});
