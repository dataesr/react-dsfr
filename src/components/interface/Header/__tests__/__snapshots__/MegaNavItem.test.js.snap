// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MegaNavItem /> renders correctly 1`] = `
<li
  className="fr-nav__item"
>
  <button
    aria-controls="xyx"
    aria-expanded={false}
    aria-label="ouvrir la navigation"
    className="fr-nav__btn"
    onClick={[Function]}
    type="button"
  >
    title
  </button>
  <div
    className="fr-mega-menu fr-collapse"
    id="xyx"
    style={
      Object {
        "--collapse": "0px",
        "maxHeight": null,
        "padding": 0,
      }
    }
  >
    <div
      className="fr-container fr-container--fluid fr-container-lg"
    >
      <button
        aria-controls="mega-menu-775"
        className="fr-link--close fr-link"
        onClick={[Function]}
        type="button"
      >
        close
      </button>
      <div
        className="fr-grid-row fr-grid-row-lg--gutters"
      >
        <div
          className="fr-col-12 fr-col-lg-8 fr-col-offset-lg-4--right"
        >
          <div
            className="fr-mega-menu__leader"
          >
            <h6
              className="fr-h4 fr-mb-2v"
            >
              title
            </h6>
            <p
              className="fr-hidden fr-displayed-lg"
            >
              description
            </p>
          </div>
        </div>
        child
      </div>
    </div>
  </div>
</li>
`;
